sql/src/main/java/org/apache/druid/sql/calcite/rel/QueryMaker.java

```diff
@@ -98,6 +100,12 @@ public ObjectMapper getJsonMapper()
   {
     final Query query = druidQuery.getQuery();
 
+    final Query innerMostQuery = findInnerMostQuery(query);
+    if (plannerContext.getPlannerConfig().isForceTimeCondition() &&
+        innerMostQuery.getIntervals().equals(Intervals.ONLY_ETERNITY)) {
+      throw new CannotBuildQueryException("Missing __time filter");
```

Comment:
Suggest having this error message mention the time condition property, so that users who encounter it know that the problem is with their query and not an internal server error.

e.g. "requireTimeCondition is enabled, all queries must include a filter condition on the __time column."